### PR TITLE
Refactor and improve tests

### PR DESCRIPTION
**Pull request by xtreamwayz**
This PR is based on #101 

- [x] Parse composer package and populate definitions and requirements
- [x] Add test for cleaning up the de requirements
- [x] Remove installPackage method from the test case and use the function from the installer
- [x] Test 'n' answers
- [x] Test answers with custom packages
- [x] Update travis file
- [x] Drop php 5.5
- [x] Add coveralls support
- [x] Improve coverage

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,27 +2,55 @@ sudo: false
...
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs --no-scripts"
+
+notifications:
+  irc: "irc.freenode.org#zftalk.dev"
+  email: false
+
 matrix:
   fast_finish: true
   include:
-    - php: 5.5
```
Comment (weierophinney) [inline .travis.yml]: If we do this for master, we need to keep 5.5 in here; if we do it for develop, and the 1.1 release, we can leave as-is. I'm going to leave it to you, @xtreamwayz , to decide which you want to target.

**Comment by weierophinney** [comment]
Awesome work, @xtreamwayz !

One note: can you please rebase? I merged a few patches from you earlier today that affect the changes you've done (notably: ocramius/proxy-manager removal from `$devDependencies`, and the looping for the "minimal install" prompt).

Thanks!

**Comment by xtreamwayz** [comment]
@weierophinney Rebase done.

File: test/ExpressiveInstallerTest/InstallerTestCase.php [hunk 2]
```diff
@@ -24,79 +26,111 @@
 class InstallerTestCase extends \PHPUnit_Framework_TestCase
```
Comment (weierophinney) [inline test/ExpressiveInstallerTest/InstallerTestCase.php]: I'm going to mark this as abstract when I merge, as the intention is for test cases to extend it.

Comment (xtreamwayz) [inline test/ExpressiveInstallerTest/InstallerTestCase.php]: I guess you can rename it to AbstractInstallerTestCase as well.

**Pull request merged**